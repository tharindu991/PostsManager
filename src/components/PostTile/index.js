/*
 * Author: Tharindu Seneviratne
 * Date: 29/02/2020
 *
 * single post with information
 */
import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {StyleSheet} from 'react-native';
import CommonStyles from '../../theme/commonStyles';
import PropTypes from 'prop-types';

const PostTile = ({username, title, onPress, testID}) => {
  return (
    <View style={[styles.container, CommonStyles.contentCenter]}>
      <TouchableOpacity
        testID={testID}
        style={[styles.touchable, styles.touchableShadow]}
        onPress={onPress}>
        <Text style={CommonStyles.boldText}>{username}</Text>
        <Text style={CommonStyles.standardText}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...CommonStyles.pageBgColor,
    flex: 1,
    paddingTop: 15,
    padding: 10,
  },
  touchable: {
    width: '100%',
    borderRadius: 6,
    padding: 20,
    backgroundColor: 'white',
  },
  touchableShadow: {
    ...CommonStyles.tileShadowColor,
    elevation: 5,
    shadowOffset: {height: 6},
    shadowOpacity: 0.3,
    shadowRadius: 4,
  },
});

PostTile.propTypes = {
  username: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default PostTile;
