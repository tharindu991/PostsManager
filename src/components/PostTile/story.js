/*
 * Author: Tharindu Seneviratne
 * Date: 29/02/2020
 *
 * Defines stories for PostTile component
 */

import React from 'react';
import {storiesOf} from '@storybook/react-native';
import PostTile from './';

storiesOf('PostTile', module).add('default view', () => (
  <PostTile
    username={'test username'}
    title={'test title'}
    onPress={() => console.log('onPress: PostTile onRefresh called')}
  />
));
