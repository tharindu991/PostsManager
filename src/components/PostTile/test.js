/*
 * Author: Tharindu Seneviratne
 * Date: 29/02/2020
 *
 * Defines tests for PostTile component
 */

import React from 'react';
import PostTile from './';
import renderer from 'react-test-renderer';
const clickFn = jest.fn();

describe('PostTile Component', () => {
  it('PostTile component renders correctly', () => {
    const component = renderer
      .create(
        <PostTile
          username={'test username'}
          title={'test title'}
          onPress={clickFn}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
