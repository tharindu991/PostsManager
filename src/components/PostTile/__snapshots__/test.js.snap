// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTile Component PostTile component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#d2d5d6",
        "flex": 1,
        "padding": 10,
        "paddingTop": 15,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 6,
        "elevation": 5,
        "opacity": 1,
        "padding": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 6,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      test username
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      test title
    </Text>
  </View>
</View>
`;
